Meta/gn/secondary/Userland/Libraries/LibWeb/generate_idl_bindings.gni

```diff
@@ -38,320 +41,87 @@
-
-  _invoke_bindings_generator(interface_name) {
-    type = "global-mixin"
-    type_filename_fragment = "GlobalMixin"
-    name = interface_name
-    path = invoker.path
-  }
-
-  source_set(interface_name + "_sources") {
-    deps = [
-      ":" + interface_name + "_prototype_sources",
-      ":" + interface_name + "_constructor_sources",
-      ":" + interface_name + "_global_mixin_sources",
-    ]
-  }
-
-  source_set("generate_" + interface_name) {
-    deps = [
-      ":generate_" + interface_name + "_prototype",
-      ":generate_" + interface_name + "_constructor",
-      ":generate_" + interface_name + "_global_mixin",
-    ]
-  }
-}
-
-# FIXME: Deduplicate these templates
-template("_bind_web_interface") {
-  forward_variables_from(invoker,
-                         [
-                           "configs",
-                           "inputs",
-                           "include_dirs",
-                           "outputs",
-                           "public_configs",
-                           "testonly",
-                           "visibility",
-                         ])
-
-  interface_name = target_name
-
-  _invoke_bindings_generator(interface_name) {
-    type = "prototype"
-    type_filename_fragment = "Prototype"
-    name = interface_name
-    path = invoker.path
-  }
-
-  _invoke_bindings_generator(interface_name) {
-    type = "constructor"
-    type_filename_fragment = "Constructor"
-    name = interface_name
-    path = invoker.path
-  }
-
-  source_set(interface_name + "_sources") {
-    deps = [
-      ":" + interface_name + "_prototype_sources",
-      ":" + interface_name + "_constructor_sources",
-    ]
-  }
-
-  source_set("generate_" + interface_name) {
-    deps = [
-      ":generate_" + interface_name + "_prototype",
-      ":generate_" + interface_name + "_constructor",
-    ]
-  }
-}
-
 template("generate_idl_bindings") {
   forward_variables_from(invoker,
                          [
                            "type",
                            "idl_list",
+                           "include_dirs",
                          ])
   idl_sources = []
   generate_idl = []
+  gen_dir = get_label_info("//Userland/Libraries/LibWeb", "target_gen_dir") +
+            "/Bindings/"
   foreach(idl, idl_list) {
+    out_name = get_path_info(idl, "name")
     path = get_path_info(rebase_path(idl, "//Userland/Libraries/LibWeb"),
-                         "dir") + "/" + get_path_info(idl, "name")
+                         "dir") + "/" + out_name
     name = string_replace(path, "/", "_")
-    if (type == "standard") {
-      _bind_web_interface(name) {
-        path = path
-      }
-    } else if (type == "iterable") {
-      _bind_web_iterable_interface(name) {
-        path = path
-      }
-    } else if (type == "namespace") {
-      _bind_web_namespace(name) {
-        path = path
-      }
+
+    output_files = []
+    if (type == "namespace") {
+      output_files += [
+        gen_dir + out_name + "Namespace.h",
```

Comment:
This is good as-is, but FYI it could also be written `"${gen_dir}${out_name}Namespace.h`

Reply:
sure, I like the way that looks better